data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -47,73 +51,118 @@
+    this.httpsServerOptions = httpsServerOptions;
     this.ingressProducerStoreFactory = ingressProducerStoreFactory;
     this.ingressRequestHandler = ingressRequestHandler;
   }
 
-
   @Override
   public void start(final Promise<Void> startPromise) {
     this.ingressProducerStore = this.ingressProducerStoreFactory.apply(vertx);
     this.messageConsumer = ResourcesReconciler
-      .builder()
-      .watchIngress(IngressReconcilerListener.all(this.ingressProducerStore, this.ingressRequestHandler))
-      .buildAndListen(vertx);
+        .builder()
+        .watchIngress(IngressReconcilerListener.all(this.ingressProducerStore, this.ingressRequestHandler))
+        .buildAndListen(vertx);
+
+    this.httpServerOptions
+        .setSsl(false);
+
+    this.httpServer = vertx.createHttpServer(this.httpServerOptions);
+
+    // check whether the secret volume is mounted
+    File secretVolume = new File("/etc/receiver-secret-volume");
+    if (secretVolume.exists()) {
+      // The secret volume is mounted, we should start the https server
+      // check whether the tls.key and tls.crt files exist
+      File tlsKeyFile = new File("/etc/receiver-secret-volume/tls.key");
+      File tlsCrtFile = new File("/etc/receiver-secret-volume/tls.crt");
 
-    this.server = vertx.createHttpServer(httpServerOptions);
+      if (tlsKeyFile.exists() && tlsCrtFile.exists() && httpsServerOptions != null) {
+        PemKeyCertOptions keyCertOptions = new PemKeyCertOptions()
+            .setKeyPath("/etc/receiver-secret-volume/tls.key")
+            .setCertPath("/etc/receiver-secret-volume/tls.crt");
+        this.httpsServerOptions
+            .setSsl(true)
+            .setPemKeyCertOptions(keyCertOptions);
 
+        this.httpsServer = vertx.createHttpServer(this.httpsServerOptions);
+
+      }
+
+    }
     final var handler = new ProbeHandler(
-      env.getLivenessProbePath(),
-      env.getReadinessProbePath(),
-      new MethodNotAllowedHandler(this)
-    );
-
-    this.server.requestHandler(handler)
-      .exceptionHandler(startPromise::tryFail)
-      .listen(httpServerOptions.getPort(), httpServerOptions.getHost())
-      .<Void>mapEmpty()
-      .onComplete(startPromise);
+        env.getLivenessProbePath(),
+        env.getReadinessProbePath(),
+        new MethodNotAllowedHandler(this));
+
+    if (this.httpsServer != null) {
+      CompositeFuture.all(
+          this.httpServer.requestHandler(handler)
+              .exceptionHandler(startPromise::tryFail)
+              .listen(this.httpServerOptions.getPort(), this.httpServerOptions.getHost())
+              .<Void>mapEmpty(),
+
+          this.httpsServer.requestHandler(handler)
+              .exceptionHandler(startPromise::tryFail)
+              .listen(this.httpsServerOptions.getPort(), this.httpsServerOptions.getHost())
+              .<Void>mapEmpty())
+          .<Void>mapEmpty().onComplete(startPromise);
```

Comment:
```suggestion
      CompositeFuture.all(
          this.httpServer.requestHandler(handler)
              .exceptionHandler(startPromise::tryFail)
              .listen(this.httpServerOptions.getPort(), this.httpServerOptions.getHost()),

          this.httpsServer.requestHandler(handler)
              .exceptionHandler(startPromise::tryFail)
              .listen(this.httpsServerOptions.getPort(), this.httpsServerOptions.getHost()))
          .<Void>mapEmpty().onComplete(startPromise);
```

mapEmpty shouldn't be needed everywhere?